mavros/src/plugins/imu.cpp

```diff
@@ -0,0 +1,530 @@
+
+		if (!m_uas->is_ardupilotmega()) {
+			ROS_WARN_THROTTLE_NAMED(60, "imu", "IMU: linear acceleration on RAW_IMU known on APM only.");
+			ROS_WARN_THROTTLE_NAMED(60, "imu", "IMU: ~imu/data_raw stores unscaled raw acceleration report.");
+			linear_accel_vec_enu.setZero();
+			linear_accel_vec_ned.setZero();
+		}
+
+		/** Magnetic field data:
+		 *  @snippet src/plugins/imu.cpp mag_field
+		 */
+		// [mag_field]
+		auto mag_field = ftf::transform_frame_aircraft_baselink<Eigen::Vector3d>(
+					Eigen::Vector3d(imu_raw.xmag, imu_raw.ymag, imu_raw.zmag) * MILLIT_TO_TESLA);
+		// [mag_field]
+
+		publish_mag(header, mag_field);
+	}
+
+	/**
+	 * @brief Handle SCALED_IMU MAVlink message.
+	 * Message specification: http://mavlink.org/messages/common/#SCALED_IMU
+	 * @param[in] msg		Received Mavlink msg
+	 * @param[in,out] imu_raw	SCALED_IMU msg
+	 */
+	void handle_scaled_imu(const mavlink::mavlink_message_t *msg, mavlink::common::msg::SCALED_IMU &imu_raw)
+	{
+		if (has_hr_imu)
+			return;
+
+		ROS_INFO_COND_NAMED(!has_scaled_imu, "imu", "IMU: Scaled IMU message used.");
+		has_scaled_imu = true;
+
+		auto imu_msg = boost::make_shared<sensor_msgs::Imu>();
+		auto header = m_uas->synchronized_header(frame_id, imu_raw.time_boot_ms);
+
+		auto gyro = ftf::transform_frame_aircraft_baselink<Eigen::Vector3d>(
+					Eigen::Vector3d(imu_raw.xgyro, imu_raw.ygyro, imu_raw.zgyro) * MILLIRS_TO_RADSEC);
+		auto accel_ned = Eigen::Vector3d(Eigen::Vector3d(imu_raw.xacc, imu_raw.yacc, imu_raw.zacc) * MILLIG_TO_MS2);
+		auto accel_enu = ftf::transform_frame_aircraft_baselink<Eigen::Vector3d>(accel_ned);
+
+		publish_imu_data_raw(header, gyro, accel_enu, accel_ned);
+
+		/** Magnetic field data:
+		 *  @snippet src/plugins/imu.cpp mag_field
+		 */
+		auto mag_field = ftf::transform_frame_aircraft_baselink<Eigen::Vector3d>(
+					Eigen::Vector3d(imu_raw.xmag, imu_raw.ymag, imu_raw.zmag) * MILLIT_TO_TESLA);
+
+		publish_mag(header, mag_field);
+	}
+
+	/**
+	 * @brief Handle SCALED_PRESSURE MAVlink message.
+	 * Message specification: http://mavlink.org/messages/common/#SCALED_PRESSURE
+	 * @param[in] msg	Received Mavlink msg
+	 * @param[in,out] press	SCALED_PRESSURE msg
+	 */
+	void handle_scaled_pressure(const mavlink::mavlink_message_t *msg, mavlink::common::msg::SCALED_PRESSURE &press)
+	{
+		if (has_hr_imu)
+			return;
+
+		auto header = m_uas->synchronized_header(frame_id, press.time_boot_ms);
+
+		auto temp_msg = boost::make_shared<sensor_msgs::Temperature>();
+		temp_msg->header = header;
+		temp_msg->temperature = press.temperature / 100.0;
+		temp_pub.publish(temp_msg);
+
+		auto atmp_msg = boost::make_shared<sensor_msgs::FluidPressure>();
+		atmp_msg->header = header;
+		atmp_msg->fluid_pressure = press.press_abs * 100.0;
+		press_pub.publish(atmp_msg);
+	}
+
+	/**
+	 * @brief Checks for connection and overrides variable values
+	 * @param[in] connected		Is connected?
+	 */
+	void connection_cb(bool connected) override
```

Comment:
That might be documented once in base class.

Reply:
what do you mean?

Reply:
Oh ok this is an override. So should I remove the associated doc tags?

Reply:
https://github.com/mavlink/mavros/blob/master/mavros/include/mavros/mavros_plugin.h you may full-fill comment in base class.
All overrides should inherit that explanation. Else we will copy-paste same explanations in each plugin.